quantum/process_keycode/process_rgb.c

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2019
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "process_rgb.h"
+#include "rgb.h"
+
+typedef void (*rgb_func_pointer)(void);
+
+/**
+ * Wrapper for inc/dec rgb keycode
+ * 
+ * noinline to optimise for firmware size not speed (not in hot path) 
+ */
+void __attribute__((noinline)) handleKeycodeRGB(const uint8_t is_shifted, const rgb_func_pointer inc_func, const rgb_func_pointer dec_func) {
```

Comment:
let's change these to return boolean and return false. This will make the caller slightly cleaner.

Reply:
So i gave the above suggestion a go, and it consumes about an extra 50 bytes with the `return false` and `return handleKeycodeRGB(`. While its about 0.17% of the available flash, we recent had the issue with trying to claw back 4 bytes. Do we want to focus on readability here? (its still a net saving)

Reply:
While I'm all for regaining as much program space as possible... I think that readability (and maintainability) should be the priority.

Reply:
If the handleKeycodeRGB had more logic, i would agree, but if it just returns false i would say personally not any more readable. That being said, i don't mind either way.

In the case of the animations, you end up with
```c
#ifdef RGBLIGHT_EFFECT_KNIGHT
                return handleKeycodeRGBMode(RGBLIGHT_MODE_KNIGHT, RGBLIGHT_MODE_KNIGHT_end);
#else
                return false;
#endif
```
or

```c
#ifdef RGBLIGHT_EFFECT_KNIGHT
                return handleKeycodeRGBMode(RGBLIGHT_MODE_KNIGHT, RGBLIGHT_MODE_KNIGHT_end);
#endif
                return false;
```

with the dead double return that gets optimised out, which in both cases is less readable to me.

Reply:
I deliberately didn't ask you to modify RGBMode because I didn't think it helped readability

Reply:
> So i gave the above suggestion a go, and it consumes about an extra 50 bytes with the `return false` and `return handleKeycodeRGB(`. While its about 0.17% of the available flash, we recent had the issue with trying to claw back 4 bytes. Do we want to focus on readability here? (its still a net saving)

I actually think that making this signature return something maybe useful in the long run, much like we had with led settings. Right now it only returns false, but there may be cases in the future where it's useful to return true.

Reply:
Sorry, I read "these" as both the new function signatures.

As this is non user facing, would we not just defer to the point we implement the functionality?

Reply:
I think that's a fair point. I still think it's slightly more readable than having an explicit false, but it's also anticipating use cases we don't have, so just leave it as is for now.